chapters/en/slides/module4/module4_02.Rmd

```diff
@@ -0,0 +1,567 @@
+but appears not to be in this plot.
+
+Note that we the color title to `None` to save some space between the subplots,
+and since it is the same for each color bar ("Count of records").
+
+To get more resolution of the counts for each column pair,
+we resolve the colorscale to be set individually for each subplot.
+Otherwise the high counts in the few bins of diagonals
+would drown out the counts in the other plots
+which would have the same color almost everywhere.
+
+TODO I am guessing it is too complicated to show df.corr() and talk about pearson and spearman here?
+
+---
+
+## Repeat also allows us to explore the relationship between categorical and numerical columns
+
+```{python}
+categorical_columns = movies_extended.select_dtypes('object').columns.tolist()
+categorical_columns = [col for col in categorical_columns if col != 'Title']
+(alt.Chart(movies_extended.drop(columns='Title'))
+ .mark_boxplot().encode(
+     alt.X(alt.repeat('column'), type='quantitative'),
+     alt.Y(alt.repeat('row'), type='nominal'))
+ .repeat(column=numerical_columns, row=categorical_columns))
+```
+
+Notes:
+In addition to repeating charts with numerical data only,
+we can use the same principles to explore the relationships
+between the categorical and the numerical columns in our dataset.
+
+Here,
+we spread the three categorical variables out along the columns of the plot
+to be able to answer questions regarding how the different categories compare,
+e.g. which genre has the longest running movies?
+
+The categories labeled "null" are the ones which are missing a value in that categorical column in the dtaframe,
+but still has values in the numiercal column that is plotted on the x-axis.
+
+We have also removed the "Title" item from the list of categorical columns,
+as well as dropped it from the dataframe,
+since it would become a giant plot if all thuosands titles were included without limiting the plot height.
+sort the columns according to their 
+
+So what is the answer to our question,
+which genre has the longest running movies?
+It seems like Westerns, Musicals, and Dramas have the longest run times
+and we can also see that for the other plots
+it is Dramatization and Historical Fcition
+as well as PG-13 an R rated movies tha run the longest.
+
+You might have noticed that it was a bit hard to compare the boxes because they were not sorted.
+Since Altair requires a list of names to sort boxplots,
+we would have needed to pass a list of all the names from all the categorical columns,
+which would have looked like this:
+
+```python
+running_time_order = []
+for groupby_col in ['Major Genre', 'Creative Type', 'MPAA Rating']:
+    running_time_order.extend(
+        movies_extended
+        .groupby(groupby_col)
+        .median()
+        ['Running Time min']
+        .sort_values()
+        .index
+        .to_list())
+...
+alt.Y(alt.repeat('row'), type='nominal', sort=running_time_order))
+...
+```
```

Comment:
Can we do this on the next slide?

Reply:
I see this as more of an extracurricular item, which we can mention is is the transcipt for those who are inclined to read it